supyr_struct/buffer.py

```diff
@@ -206,20 +243,31 @@ def seek(self, pos, whence=SEEK_SET):
         Raises ValueError if whence is not SEEK_SET, SEEK_CUR, or SEEK_END.
         Raises TypeError if whence is not an int.
         '''
+        # TODO: should these asserts be replaced with raise?
         if whence == SEEK_SET:
-            self[pos - 1]  # check if seek is outside of range
             assert pos >= 0, "Read position cannot be negative."
+
+            if pos - 1 not in range(len(self)):
+                raise IndexError('seek position out of range')
+
             self._pos = pos
         elif whence == SEEK_CUR:
-            self[self._pos + pos - 1]  # check if seek is outside of range
-            assert self._pos + pos >= 0, "Read position cannot be negative."
-            self._pos += pos
+            pos = self._pos + pos
+            assert pos >= 0, "Read position cannot be negative."
+
+            if pos - 1 not in range(len(self)):
+                raise IndexError('seek position out of range')
+
+            self._pos = pos
         elif whence == SEEK_END:
```

Comment:
The way this was done was for speed purposes. If this way is faster, then by all means go ahead. The comparison of type directly to int instead of isinstance was for speed. So was accessing an item to check being able to seek to a spot rather than not in range(len(self)). The asserts should definitely be replaced with exception raising though.